.NET/Microsoft.Recognizers.Text.DataDrivenTests/TestHelpers.cs

```diff
@@ -0,0 +1,313 @@
+                case "DateTimeModel":
+                    return DateTimeRecognizer.GetInstance(DateTimeOptions.None).GetDateTimeModel(language);
+                case "CustomNumberModel":
+                    return GetCustomModelFor(language);
+            }
+            return null;
+        }
+
+        public static IExtractor GetExtractor(this TestContext context)
+        {
+            var language = TestUtils.GetCulture(context.FullyQualifiedTestClassName);
+            var extractorName = TestUtils.GetExtractorName(context.TestName);
+            switch (language)
+            {
+                case Culture.English:
+                    return GetEnglishExtractor(extractorName);
+                case Culture.Spanish:
+                    return GetSpanishExtractor(extractorName);
+            }
+            return null;
+        }
+
+        public static IDateTimeParser GetDateTimeParser(this TestContext context)
+        {
+            var language = TestUtils.GetCulture(context.FullyQualifiedTestClassName);
+            var parserName = TestUtils.GetParserName(context.TestName);
+            switch (language)
+            {
+                case Culture.English:
+                    return GetEnglishParser(parserName);
+                case Culture.Spanish:
+                    return GetSpanishParser(parserName);
+            }
+            return null;
+        }
+
+        public static IExtractor GetEnglishExtractor(string extractorName)
+        {
+            switch (extractorName)
+            {
+                case "Date":
+                    return new BaseDateExtractor(new EnglishDateExtractorConfiguration());
+                case "Time":
+                    return new BaseTimeExtractor(new EnglishTimeExtractorConfiguration());
+                case "DatePeriod":
+                    return new BaseDatePeriodExtractor(new EnglishDatePeriodExtractorConfiguration());
+                case "TimePeriod":
+                    return new BaseTimePeriodExtractor(new EnglishTimePeriodExtractorConfiguration());
+                case "DateTime":
+                    return new BaseDateTimeExtractor(new EnglishDateTimeExtractorConfiguration());
+                case "DateTimePeriod":
+                    return new BaseDateTimePeriodExtractor(new EnglishDateTimePeriodExtractorConfiguration());
+                case "Duration":
+                    return new BaseDurationExtractor(new EnglishDurationExtractorConfiguration());
+                case "Holiday":
+                    return new BaseHolidayExtractor(new EnglishHolidayExtractorConfiguration());
+                case "Set":
+                    return new BaseSetExtractor(new EnglishSetExtractorConfiguration());
+                case "Merged":
+                    return new BaseMergedExtractor(new EnglishMergedExtractorConfiguration(), DateTimeOptions.None);
+                case "MergedSkipFromTo":
```

Comment:
These case values are now test-case-specific, right? Shouldn't we have something like a string-backed enum to keep them?
I'm not fond of the idea where devs needs to remember specific strings and if we decide to rename them, if will affect many different places.

Reply:
Updated to use Enums instead of strings